server/main.go

```diff
@@ -37,73 +37,15 @@ var reCaptchaSecret string
 var rpcUrl string
 var logger *zap.Logger
 
-const defaultFetchLimit = 500
-
-func parseTime(r *http.Request) (time.Time, time.Time, error) {
-	var err error
-	fromTime := time.Unix(0, 0)
-	toTime := time.Now()
-	fromTimeStr := r.URL.Query().Get("from_time")
-	toTimeStr := r.URL.Query().Get("to_time")
-	if fromTimeStr != "" {
-		fromTime, err = time.Parse(time.RFC3339, fromTimeStr)
-		if err != nil {
-			return fromTime, toTime, err
-		}
-	}
-	if toTimeStr != "" {
-		toTime, err = time.Parse(time.RFC3339, toTimeStr)
-		if err != nil {
-			return fromTime, toTime, err
-		}
-	}
-	return fromTime, toTime, nil
-}
-
-func parseBool(r *http.Request, name string) (boolean *bool) {
-	valStr := r.URL.Query().Get(name)
-	var val bool
-	if valStr != "" {
-		if valStr == "true" {
-			val = true
-		} else {
-			val = false
-		}
-		return &val
+func parseGetParam(r *http.Request, w http.ResponseWriter, result interface{}) error {
+	if err := schema.NewDecoder().Decode(result, r.URL.Query()); err != nil {
+		logger.Info("failed to parse get params")
+		errorResponse(w, http.StatusBadRequest, errors.New("invalid params"))
```

Comment:
Should the err be used so they know what's wrong? And the info log probably can be removed since the request will already be logged.